src/hotspot/os/windows/semaphore_windows.cpp

```diff
@@ -39,7 +39,7 @@ WindowsSemaphore::~WindowsSemaphore() {
   ::CloseHandle(_semaphore);
 }
 
-void WindowsSemaphore::signal(uint count) {
+void WindowsSemaphore::signal(uint count, bool ignore_overflow) {
```

Comment:
I believe the Windows ReleaseSemaphore can also fail with the equivalent of EOVERFLOW if the count would exceed the maximum allowed. We either need to dig deeper into what value of GetlastError we would see, or else ignore all errors if ignore_overflow is set. I think I can tweak a gtest to see what happens.